extensions/ql-vscode/src/vscode-tests/cli-integration/remote-queries/run-remote-query.test.ts

```diff
@@ -334,7 +334,7 @@ describe('Remote queries', function() {
       const fileUri = getFile('data-remote-qlpack-nested/subfolder/in-pack.ql');
 
       const querySubmissionResult = await runRemoteQuery(cli, credentials, fileUri, true, progress, token);
-      expect(querySubmissionResult).to.be.ok;
+      expect(querySubmissionResult).to.be.undefined;
```

Comment:
Alternatively I can see in other places we're calling this method without expecting a result to be returned. If that's the new behaviour, I think we should change the tests to stop expecting a result in the first place. So instead of:

```
const querySubmissionResult = await runRemoteQuery(cli, credentials, fileUri, true, progress, token);
```

we'd just call the method: 
```
await runRemoteQuery(cli, credentials, fileUri, true, progress, token);
```

Reply:
I was kinda putting this off for this PR and hoping to address it in the future when we know better what data we'll want returned from this method. But I think we can actually just return a `VariantAnalysis` object and that likely be exactly what we want, or at least it's a good starting point.

So I've adjusted `RemoteQuerySubmissionResult` to looks like the following so it can handle the live and non-live results cases.
```typescript
export interface RemoteQuerySubmissionResult {
  queryDirPath?: string;
  query?: RemoteQuery;
  variantAnalysis?: VariantAnalysis;
}
```

In the tests we can now assert that the object is non-undefined. I've gone for only checking a couple of fields on it. We could assert the values of more field but I think this is good enough for now?

Here's what the logging looks like. I expect we'll change this in the future now but this is a placeholder until we do more.
<img width="1213" alt="Screenshot 2022-09-27 at 10 14 31" src="https://user-images.githubusercontent.com/3749000/192486870-7c3f601f-800f-498f-92b5-33b2c3ac096b.png">

Reply:
Nice! Thanks for changing it 👍